### PR TITLE
Add CI step for publishing `sway-lib-std` docs

### LINKED ISSUE #3549: Publish output from `forc doc` for `sway-lib-std` on ghpages
As we refactor CI to publish both "The Sway Book" and "The Sway Reference" under ghpages (rather than just the former), we should also add a std reference generated by `forc doc`.

In the long-run, we can take some inspiration from docs.rs and automate the generation of all pkg documentation from the hypothetical future package registry. #865

### PR DESCRIPTION
**Pull request by eureka-cpu**
Closes #3549 

Installs `forc-doc`, runs `forc doc --manifest-path ./sway-lib-std` to produce the docs, then publishes from `./sway-lib-std/out/doc` where the root index can be found.

### CODE REVIEW

File: .github/workflows/gh-pages.yml [hunk 1]
```diff
@@ -46,6 +47,9 @@ jobs:
       - name: Build Sway reference
         run: mdbook build docs/reference
 
+      - name: Build Sway std library
+        run: forc doc --manifest-path ./sway-lib-std && cp sway-lib-std/out/doc/* docs/std
```
Comment (bingcicle) [inline .github/workflows/gh-pages.yml:51]:
```suggestion
        run: forc doc --manifest-path ./sway-lib-std && cp -r sway-lib-std/out/doc/* docs/std
```

File: .github/workflows/gh-pages.yml [hunk 2]
```diff
@@ -62,6 +66,14 @@ jobs:
           destination_dir: master/reference
         if: github.ref == 'refs/heads/master'
 
+      - name: Deploy master std
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          publish_dir: ./docs/std/book
```
Comment (bingcicle) [inline .github/workflows/gh-pages.yml:73]: `publish_dir` is the source directory, so actually something like 
```suggestion
          publish_dir: ./sway-lib-std/out
```

_could_ work.

File: .github/workflows/gh-pages.yml (hunk 1 above)
Comment (eureka-cpu) [reply .github/workflows/gh-pages.yml:51]: ah nice catch

Comment (eureka-cpu) [reply .github/workflows/gh-pages.yml:51]: oh your below comment would basically negate this

File: .github/workflows/gh-pages.yml (hunk 2 above)
Comment (eureka-cpu) [reply .github/workflows/gh-pages.yml:73]: wouldn't we want this to be `./sway-lib-std/out/doc` since we don't want all of the out directory?

**Comment by eureka-cpu** [comment]
Working on the foundation [here](https://github.com/eureka-cpu/std-ci-test), still trouble-shooting

**Comment by bingcicle** [comment]
> Working on the foundation [here](https://github.com/eureka-cpu/std-ci-test), ~still trouble-shooting~
> 
> It's working now at https://eureka-cpu.github.io/std-ci-test/master/
> 
> Should be up soon

That looks good :exploding_head:

**Comment by mohammadfawaz** [comment]
@eureka-cpu remind me is there an issue tracking the addition of methods implemented for types to the docs? For example, the method `insert` for `StorageMap`, etc.

**Comment by eureka-cpu** [comment]
> @eureka-cpu remind me is there an issue tracking the addition of methods implemented for types to the docs? For example, the method `insert` for `StorageMap`, etc.

Ah thanks for the reminder. I have it in my TODOs in the code but not in an actual issue. Filing now!

**Review by mitchmindtree — APPROVED** [review]
lgtm!

**Review by bingcicle — APPROVED** [review]

**Pull request merged**